### ISSUE TITLE
Regression from v`1.0.0`

### ISSUE THREAD

**Issue by Rudxain** [frewsxcv/rust-dark-light#39]
Universal-Debloater-Alliance/universal-android-debloater-next-generation#627

`src/main.rs`:
```rust
fn main() {
    println!("{:?}", dark_light::detect());
}
```

`Cargo.toml`:
```toml
[package]
name = "tmp"
version = "0.1.0"
edition = "2021"
```
```toml
[dependencies]
dark-light = "=1.1.1"
# `cargo run`: "Light"
```
```toml
[dependencies]
dark-light = "=1.0.0"
# `cargo run`: "Dark" (correct)
```

The 2 variations of `dependencies` from `Cargo.toml` where tested on **separate directories**, for better isolation (no need for `cargo update`) and to avoid recompilation of recursive deps (although I tried to recycle `target/`)

**Comment by maplant** [comment]
I experience this as well, dark-light version 1.1.1 returns an incorrect theme for detect

**Comment by jacksongoode** [comment]
Can you narrow down if it was from 1.1.0 to 1.1.1 or 1.0.0 to 1.1.0. It could be the zbus bump which caused this.

**Comment by maplant** [comment]
> Can you narrow down if it was from 1.1.0 to 1.1.1 or 1.0.0 to 1.1.0. It could be the zbus bump which caused this.

It was `1.1.0` that introduced this issue

**Comment by klassenserver7b** [comment]
Yes, it is zbus bump because zbus 0.4 introduced `body.deserialize::<T>(). This code uses:
```RUST
let theme = reply.body().deserialize::<u32>();
```
But the body of the message is structured like this: `Structure { fields: [Value(Value(U32(1)))], signature: Signature("(v)") }`
You can check this yourself by parsing the reply like that:
```RUST
        let body = reply.body();
        let theme: zbus::zvariant::Structure = body.deserialize().unwrap();
        println!("theme: {:?}", theme);
```

**Comment by klassenserver7b** [comment]
Therefore i would recommend to use `ReadOne` to get rid of one Value()
```RUST
    let reply = conn.unwrap().call_method(
        Some("org.freedesktop.portal.Desktop"),
        "/org/freedesktop/portal/desktop",
        Some("org.freedesktop.portal.Settings"),
        "ReadOne",
        &("org.freedesktop.appearance", "color-scheme"),
    );
    ```
    
    After that is is required to downcast the Value(U32) to U32.
    This would be working code:
    
    ```RUST
     let body = reply.body();
        let value = body.deserialize::<Value>();
        if value.is_err() {
            return None;
        }
        
        let theme = value.unwrap().downcast::<u32>();
        if theme.is_err() {
            return None;
        }

        match theme.unwrap() {
            1 => Some(Mode::Dark),
            2 => Some(Mode::Light),
            _ => None,
        }
        ```

**Comment by klassenserver7b** [comment]
You could also keep `Read` as dbus method. With that you would have to do the downcasting twice

**Comment by klassenserver7b** [comment]
Oh, i saw its fixed on main by using ashpd.

**Comment by edfloreshz** [comment]
Should be fixed in the next release, use main for now.

Closing this.

**Comment by Rudxain** [comment]
I re-tested the min-repo with the 2.0 release. I get interesting output:
- Dark: `Ok(Dark)`
- Light: `Ok(Unspecified)`

Is this a GNOME bug or something?

**Comment by edfloreshz** [comment]
Given that we are no longer falling back to `Mode::Light` in case of errors, that must mean that `Mode::Unspecified` comes from the system.

**Comment by edfloreshz** [comment]
@Rudxain Do you have another environment where you can test?

**Comment by Rudxain** [comment]
> @Rudxain Do you have another environment where you can test?

I currently don't have access to any Windows systems, much less MacOS. I have access to 2 PCs, both with Debian 13 (Testing channel) installed. The only "significant" difference from "vanilla Debian" is that I followed [these instructions](https://support.mozilla.org/en-US/kb/install-firefox-linux#w_install-firefox-deb-package-for-debian-based-distributions-recommended) some months ago. I've also uninstalled some "recommended" (optional dependencies) and "important" ("expected" for Unix systems), the only ones that could be related to this issue are the `*xorg*` ones. I have `xserver-common` installed because of `xwayland`
